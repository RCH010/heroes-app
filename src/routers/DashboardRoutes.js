import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar/>
            <div className='container-fluid'>
                <Switch>
                    <Route exact path='/marvel' component= { MarvelScreen} />
                    <Route exact path='/dc' component= { DcScreen } />
                    <Route exact path='/hero/:heroeId' component= { HeroScreen } />
                    <Route exact path='/search' component= {SearchScreen} />
                    <Redirect to ='/marvel'/>
                </Switch>
            </div>
        </>
    )
}
