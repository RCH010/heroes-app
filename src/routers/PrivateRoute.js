import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types';

export const PrivateRoute = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    // recordar la ruta
    localStorage.setItem('lastPath', rest.location.pathname);

    return (
        
        <Route {...rest}
            component = { (props) => (
                (isAuthenticated) 
                    ? <Component {...props} />
                    : <Redirect to='/login' />
            )}
        />
    )
}

PrivateRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}