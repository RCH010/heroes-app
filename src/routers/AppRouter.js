import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const {user} = useContext(AuthContext)

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute 
                        exact
                        component={LoginScreen}
                        isAuthenticated={user.logged}
                        path='/login'
                    />
                    {/* <Route exact path='/login' component={LoginScreen} /> */}
                    <PrivateRoute 
                        path='/' 
                        component={DashboardRoutes}
                        isAuthenticated={user.logged}
                    />
                    {/* Aquí este AppRouter es el principal, y en el caso que no sea login
                        se va y carga el DashboardRoutes. Ahí hay un switch que es el que entraría
                        aquí... */}
                </Switch>
            </div>
        </Router>
    )
}
