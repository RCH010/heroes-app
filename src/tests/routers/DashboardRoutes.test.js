import { mount } from "enzyme"
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext"
import { DashboardRoutes } from "../../routers/DashboardRoutes"

describe('Tests un <DashboardToutes />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'RaúlC'
        }
    };

    test('should display correctly', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        )
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('RaúlC');
    })
    
})
