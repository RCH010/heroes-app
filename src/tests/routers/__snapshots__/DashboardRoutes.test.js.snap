// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests un <DashboardToutes /> should display correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          RaúlC
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid"
  >
    <div>
      <h1>
        Marvel
      </h1>
      <br />
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Spider Man
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Captain America
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Iron Man
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Thor
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Hulk
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Wolverine
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Daredevil
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Hawkeye
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Cyclops
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h2
                  className="card-title mb-0 h5"
                >
                  Silver Surfer
                </h2>
                <hr
                  className="my-1"
                />
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  />
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
